### PR TITLE
1035 add supplier to incoming stock lines

### LINKED ISSUE #1035: Add supplier to incoming stock lines
When a store receives stock, the supplier should be saved into the stock lines and displayed in Stock UI

### PR DESCRIPTION
**Pull request by roxy-dao**
Closes #1035 

- Added a supplier field to stock line 
- Inbound stock saves supplier to stock line
- Applied changes to `Stock` view

Screen Shot 2023-01-18 at 11 50 42 AM

### CODE REVIEW

File: client/packages/common/src/intl/locales/en/inventory.json [hunk 1]
```diff
@@ -23,6 +23,8 @@
   "label.locked": "Locked",
   "label.batch": "Batch",
   "label.pricing": "Pricing",
+  "label.supplier": "Supplier",
+  "label.no-supplier": "Unknown",
```
Comment (roxy-dao) [inline client/packages/common/src/intl/locales/en/inventory.json:27]: Could also be `N/A` or `Inventory Adjustment`... not to sure what to label stock that appear through stocktake 🤔

File: client/packages/common/src/intl/locales/en/inventory.json [hunk 2]
```diff
@@ -23,6 +23,8 @@
   "label.locked": "Locked",
   "label.batch": "Batch",
   "label.pricing": "Pricing",
+  "label.supplier": "Supplier",
+  "label.no-supplier": "Inventory Adjustment",
```
Comment (mark-prins) [inline client/packages/common/src/intl/locales/en/inventory.json:27]: :nit the text isn't a label, it's just text.. 
I would have `"messages.no-supplier": "Inventory Adjustment",` to fit in with the other items

**Review by mark-prins — COMMENTED** [review]
Can you please add a migration for the stocktake table too? 
The `inventory_adjustment_id` column is only added in the diesel migrations, so wouldn't work for existing databases

File: server/service/src/stocktake/update.rs [hunk 3]
```diff
@@ -886,6 +894,7 @@ mod test {
             stocktake_line.sell_price_per_pack.unwrap()
         );
         assert_eq!(stock_line.note, stocktake_line.note);
+        assert_eq!(stock_line.name_id, None);
```
Comment (mark-prins) [inline server/service/src/stocktake/update.rs:897]: would be good to also have a test that links a supplier name in

File: server/cli/src/refresh_dates.rs [hunk 4]
```diff
@@ -364,6 +364,7 @@ mod tests {
                 r.item_id = mock_item_a().id;
                 r.store_id = mock_store_a().id;
                 r.expiry_date = Some(NaiveDate::from_ymd(2023, 02, 01));
+                r.name_id = Some(String::from("name_store_b"));
```
Comment (andreievg) [inline server/cli/src/refresh_dates.rs:367]: technically not needed for this test and inline_init allows for it to not be specified (keeps test data specific to test case)

**Review by andreievg — COMMENTED** [review]
thanks, a few minor comments so far:

I think it should be supplier_id rather then name_id, name_id in this case is a little bit ambiguous (especially when we start adding donor_id and manufacturer_id)

The main reason for adding inline_init helper was to reduce updates in test_data when schema changed. I saw quite a few changes there where `r.name_id = Some(donor)` was added, this is not needed.

Also another trick, server/repository/src/mock/stock_line.rs, can do:
`node util/src/to_inline_init.js repository/src/mock/stock_line.rs StockLineRow`, from server dir (see description in to_inline_init.js), it would change StockLineRow instantiation to use inline_init, at which point optionals would be automatically set to None: https://github.com/openmsupply/open-msupply/compare/example-of-using-to_inline_init

File: server/service/src/stocktake/update.rs (hunk 3 above)
Comment (roxy-dao) [reply server/service/src/stocktake/update.rs]: [2e1c1ac](https://github.com/openmsupply/open-msupply/pull/1044/commits/2e1c1ace4742de85361e40e3522665ab6a2212b1)

**Review by andreievg — APPROVED** [review]
Thank you looks good

**Pull request merged**